### PR TITLE
Enhancement/carousel v-model

### LINKED ISSUE #1358: Carousel improvements
There should be an event that fires on item change with payload that tells which item is active. 

What If I want to add caption or another components that corresponds to the active item?

### PR DESCRIPTION
**Pull request by jacekkarczmarczyk**
This PR adds model to carousel component and fixes #1358 

Playground:

```
<template>
    <v-app>
        <main>
            <v-container>
                <div class="headline">With v-model</div>
                <v-carousel v-model="model" style="height: 300px">
                    <v-carousel-item v-for="(item,i) in items" v-bind:src="item.src" :key="i"></v-carousel-item>
                </v-carousel>
                <div class="headline">Without v-model</div>
                <v-carousel style="height: 300px">
                    <v-carousel-item v-for="(item,i) in items" v-bind:src="item.src" :key="i"></v-carousel-item>
                </v-carousel>
            </v-container>
        </main>
    </v-app>
</template>

<script>
    export default {
        data () {
            return {
                model: 0,
                items: [1,2,3,4].map(i => ({
                    src: `http://lorempixel.com/400/200/?${i}`
                }))
            }
        },

        watch: {
            model () {
                console.log(`Current element: ${this.model}`)
            }
        }
    }
</script>
```

### CODE REVIEW

File: src/components/VCarousel/VCarousel.vue [hunk 1]
```diff
@@ -125,7 +121,7 @@
         this.$nextTick(this.startInterval)
       },
       init () {
-        this.current = 0
+        this.current = this.value >>> 0
```
Comment (nekosaur) [inline src/components/VCarousel/VCarousel.vue]: `this.value || 0` is more readable, and what we usually use

File: src/components/VCarousel/VCarousel.vue [hunk 2]
```diff
@@ -71,6 +62,11 @@
 
         !this.isBooted && this.cycle && this.restartInterval()
         this.isBooted = true
+
+        this.$emit('input', this.current)
+      },
+      value () {
```
Comment (nekosaur) [inline src/components/VCarousel/VCarousel.vue]: You haven't defined `value` as a prop. I would probably try to remove `current` entirely in favor of `value` and emitting `input`.

Comment (johnleider) [inline src/components/VCarousel/VCarousel.vue]: I'm in favor of this.

**Comment by jacekkarczmarczyk** [comment]
@nekosaur tests ready

File: src/components/VCarousel/VCarousel.vue (hunk 2 above)
Comment (nekosaur) [inline src/components/VCarousel/VCarousel.vue]: After talking to @jacekkarczmarczyk, I think I'm fine with keeping current, as it allows one to use the carousel without declaring a v-model. What do you think @johnleider?

Comment (johnleider) [inline src/components/VCarousel/VCarousel.vue]: I'm okay with it, can we name it inputValue though?

**Pull request merged**